File: client/phone.lua
local QBCore = exports['qb-core']:GetCoreObject()
local phoneOpen = false
local playerCitizenId = nil
local playerName = ""

-- Open phone app
RegisterNetEvent("weazel:phone:open", function()
    if phoneOpen then return end
    phoneOpen = true

    local Player = QBCore.Functions.GetPlayerData()
    playerCitizenId = Player.citizenid
    playerName = Player.charinfo.firstname .. " " .. Player.charinfo.lastname

    SetNuiFocus(true, true)
    SendNUIMessage({
        action = "phoneOpen",
        citizenid = playerCitizenId,
        playerName = playerName
    })

    TriggerServerEvent("weazel:server:getPhoneFeed")
end)
RegisterNUICallback("CustomApp", function(data, cb)
    if data.app == "weazel" and data.action == "open" then
        TriggerEvent("weazel:phone:open")
    end
    cb("ok")
end)
RegisterNUICallback("getPhoneData", function(data, cb)
    if data and data.app == "weazel" then
        cb({
            name = "weazel",
            label = "Weazel News",
            icon = "newspaper",
            ui = true
        })
    else
        cb({})
    end
end)
-- Receive feed from server
RegisterNetEvent("weazel:phone:setFeed", function(articles)
    SendNUIMessage({ action = "setFeed", articles = articles })
end)

-- NUI Callbacks
RegisterNUICallback("phoneClose", function(_, cb)
    phoneOpen = false
    SetNuiFocus(false, false)
    cb({})
end)

RegisterNUICallback("phoneGetArticle", function(data, cb)
    TriggerServerEvent("weazel:server:getArticle", data.id)
    cb({})
end)

RegisterNUICallback("phoneLikeArticle", function(data, cb)
    TriggerServerEvent("weazel:server:likeArticle", data.articleId)
    cb({})
end)

RegisterNUICallback("phoneAddComment", function(data, cb)
    TriggerServerEvent("weazel:server:addComment", data)
    cb({})
end)

RegisterNUICallback("phoneEditComment", function(data, cb)
    TriggerServerEvent("weazel:server:editComment", data)
    cb({})
end)

RegisterNUICallback("phoneDeleteComment", function(data, cb)
    TriggerServerEvent("weazel:server:deleteComment", data.commentId)
    cb({})
end)

-- Forward real-time updates to phone UI
RegisterNetEvent("weazel:client:updateLikes", function(data)
    if phoneOpen then
        SendNUIMessage({ action = "updateLikes", articleId = data.articleId, count = data.count, liked = data.liked })
    end
end)

RegisterNetEvent("weazel:client:updateComments", function(data)
    if phoneOpen then
        SendNUIMessage({ action = "updateComments", articleId = data.articleId, comments = data.comments })
    end
end)

RegisterNetEvent("weazel:client:showArticle", function(article)
    if phoneOpen then
        SendNUIMessage({ action = "showArticle", article = article })
    end
end)

File: html/phone.js
let currentView = 'feed';
let currentArticleId = null;
let playerCitizenId = null;
let playerName = '';
let editingCommentId = null;

// DOM elements
const feedView = document.getElementById('feedView');
const articleView = document.getElementById('articleView');
const backBtn = document.getElementById('backBtn');
const closeBtn = document.getElementById('closeBtn');
const feedList = document.getElementById('feedList');
const articleTitle = document.getElementById('articleTitle');
const articleMeta = document.getElementById('articleMeta');
const articleImage = document.getElementById('articleImage');
const articleContent = document.getElementById('articleContent');
const likeButton = document.getElementById('likeButton');
const likeIcon = document.getElementById('likeIcon');
const likeCount = document.getElementById('likeCount');
const commentInput = document.getElementById('commentInput');
const submitComment = document.getElementById('submitComment');
const commentsList = document.getElementById('commentsList');

// Modal
const editModal = document.getElementById('editModal');
const editCommentInput = document.getElementById('editCommentInput');
const saveEditBtn = document.getElementById('saveEditBtn');
const cancelEditBtn = document.getElementById('cancelEditBtn');

function fetchNUI(event, data) {
    fetch(`https://${GetParentResourceName()}/${event}`, {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify(data || {})
    }).catch(() => {});
}

// Show feed
function showFeed() {
    currentView = 'feed';
    feedView.style.display = 'block';
    articleView.style.display = 'none';
    backBtn.style.display = 'none';
}

// Show article
function showArticle(article) {
    currentView = 'article';
    currentArticleId = article.id;
    feedView.style.display = 'none';
    articleView.style.display = 'block';
    backBtn.style.display = 'block';

    articleTitle.innerText = article.title;
    articleMeta.innerText = `${article.author || 'Weazel'} • ${article.date || ''}`;
    
    if (article.image) {
        articleImage.innerHTML = `<img src="${article.image}" alt="Article image">`;
    } else {
        articleImage.innerHTML = '';
    }
    
    articleContent.innerText = article.content;

    // Update like button
    likeCount.innerText = article.likes || 0;
    likeIcon.innerText = article.liked ? '❤️' : '🤍';
    likeButton.style.background = article.liked ? 'rgba(192,18,46,0.2)' : 'rgba(192,18,46,0.1)';

    renderComments(article.comments || []);
}

// Render feed
function renderFeed(articles) {
    feedList.innerHTML = '';
    if (!articles || articles.length === 0) {
        feedList.innerHTML = '<p style="text-align:center; color:#8a8f99;">No articles yet.</p>';
        return;
    }
    articles.forEach(a => {
        const card = document.createElement('div');
        card.className = 'news-card';
        card.onclick = () => fetchNUI('phoneGetArticle', { id: a.id });
        card.innerHTML = `
            <h3>${a.title}</h3>
            <div class="meta">${a.author || 'Weazel'} • ${a.date || ''}</div>
            <div class="stats">
                <span>❤️ ${a.likes || 0}</span>
                <span>💬 ${a.comments || 0}</span>
            </div>
        `;
        feedList.appendChild(card);
    });
}

// Render comments
function renderComments(comments) {
    commentsList.innerHTML = '';
    if (!comments || comments.length === 0) {
        commentsList.innerHTML = '<p style="color:#8a8f99;">No comments yet.</p>';
        return;
    }
    comments.forEach(c => {
        const div = document.createElement('div');
        div.className = 'comment';
        div.innerHTML = `
            <div class="comment-header">
                <span class="comment-author">${c.author_name}</span>
                <span class="comment-date">${c.date}</span>
            </div>
            <div class="comment-text">${c.comment}</div>
        `;
        // Show edit/delete if owned by player
        if (c.citizenid === playerCitizenId) {
            const actions = document.createElement('div');
            actions.className = 'comment-actions';
            const editBtn = document.createElement('button');
            editBtn.innerText = 'Edit';
            editBtn.onclick = (e) => {
                e.stopPropagation();
                editingCommentId = c.id;
                editCommentInput.value = c.comment;
                editModal.style.display = 'flex';
            };
            const delBtn = document.createElement('button');
            delBtn.innerText = 'Delete';
            delBtn.onclick = (e) => {
                e.stopPropagation();
                if (confirm('Delete this comment?')) {
                    fetchNUI('phoneDeleteComment', { commentId: c.id });
                }
            };
            actions.appendChild(editBtn);
            actions.appendChild(delBtn);
            div.appendChild(actions);
        }
        commentsList.appendChild(div);
    });
}

// Update like button
function updateLikeButton(count, liked) {
    likeCount.innerText = count;
    likeIcon.innerText = liked ? '❤️' : '🤍';
    likeButton.style.background = liked ? 'rgba(192,18,46,0.2)' : 'rgba(192,18,46,0.1)';
}

// Event listeners
backBtn.addEventListener('click', () => {
    if (currentView === 'article') {
        showFeed();
    }
});

closeBtn.addEventListener('click', () => {
    fetchNUI('phoneClose', {});
});

likeButton.addEventListener('click', () => {
    if (currentArticleId) {
        fetchNUI('phoneLikeArticle', { articleId: currentArticleId });
    }
});

submitComment.addEventListener('click', () => {
    const text = commentInput.value.trim();
    if (text && currentArticleId) {
        fetchNUI('phoneAddComment', {
            articleId: currentArticleId,
            comment: text
        });
        commentInput.value = '';
    }
});

saveEditBtn.addEventListener('click', () => {
    const newText = editCommentInput.value.trim();
    if (newText && editingCommentId) {
        fetchNUI('phoneEditComment', {
            commentId: editingCommentId,
            comment: newText
        });
    }
    editModal.style.display = 'none';
    editingCommentId = null;
});

cancelEditBtn.addEventListener('click', () => {
    editModal.style.display = 'none';
    editingCommentId = null;
});

// Message listener
window.addEventListener('message', (event) => {
    const data = event.data;
    if (!data) return;

    switch (data.action) {
        case 'phoneOpen':
            playerCitizenId = data.citizenid;
            playerName = data.playerName;
            showFeed();
            break;
        case 'setFeed':
            renderFeed(data.articles);
            break;
        case 'showArticle':
            showArticle(data.article);
            break;
        case 'updateLikes':
            if (currentArticleId === data.articleId) {
                updateLikeButton(data.count, data.liked);
            }
            break;
        case 'updateComments':
            if (currentArticleId === data.articleId) {
                renderComments(data.comments);
            }
            break;
    }
});

// Close modal on outside click
window.addEventListener('click', (e) => {
    if (e.target === editModal) {
        editModal.style.display = 'none';
        editingCommentId = null;
    }
});

// Limit comment input to 150 chars
commentInput.addEventListener('input', function() {
    if (this.value.length > 150) {
        this.value = this.value.slice(0, 150);
    }
});
editCommentInput.addEventListener('input', function() {
    if (this.value.length > 150) {
        this.value = this.value.slice(0, 150);
    }
});